modoboa/lib/viewsets.py

```diff
@@ -0,0 +1,11 @@
+"""ViewSet related mixins and tools."""
+
+from reversion.views import RevisionMixin
+
+
+class RevisionModelMixin(RevisionMixin):
```

Comment:
Why do you declare this mixin? Why don't you use the RevisionMixin directly?

Reply:
In order to easily update it in a near future, if `RevisionMixin` evolves and does not fit anymore for example. I thought it would prevent to update each viewsets which relies on it - especially in extensions. What do you think?

Reply:
I think it makes sense. Then this PR looks ready to merge right?

Reply:
Yes I think so, also:

> - Is it a good idea to rely on reversion.views.RevisionMixin or should Modoboa implements its own mixin?
> - Is it okay to decorate all methods - e.g. not just create(), update() and delete()?

I think we can see later with time since `modoboa.lib.viewsets.RevisionModelMixin` can easily be modified thanks to that.

Reply:
Yes, I think we can make some tests with what you did and check if it behaves as we want.